velox/exec/tests/WindowTest.cpp

```diff
@@ -487,5 +487,45 @@ TEST_F(WindowTest, nagativeFrameArg) {
   }
 }
 
+DEBUG_ONLY_TEST_F(WindowTest, frameColumnNullCheck) {
+  // Null values in order-by column 's0' and frame column 'off0' do not match,
+  // so exception is expected.
+  auto inputThrow = {makeRowVector(
+      {"c0", "p0", "s0", "off0"},
+      {
+          makeNullableFlatVector<int64_t>({1, std::nullopt, 1, 2, 2}),
+          makeFlatVector<int64_t>({1, 2, 1, 2, 1}),
+          makeNullableFlatVector<int64_t>({1, 2, 3, std::nullopt, 5}),
+          makeNullableFlatVector<int64_t>({2, std::nullopt, 4, 5, 6}),
+      })};
+  auto planNode =
+      PlanBuilder(std::make_shared<core::PlanNodeIdGenerator>())
+          .values(inputThrow)
+          .window(
+              {"sum(c0) OVER (PARTITION BY p0 ORDER BY s0 RANGE BETWEEN UNBOUNDED PRECEDING AND off0 FOLLOWING)"})
+          .planNode();
+  VELOX_ASSERT_THROW(
+      AssertQueryBuilder(planNode).copyResults(pool()),
+      "Frame column should be NULL iff ORDER BY value is NULL");
+
+  // Null values in order-by column 's0' and frame column 'off0' match, so no
+  // exception should be thrown.
+  auto inputNoThrow = {makeRowVector(
+      {"c0", "p0", "s0", "off0"},
+      {
+          makeNullableFlatVector<int64_t>({1, 1, 2, std::nullopt, 2}),
+          makeFlatVector<int64_t>({1, 1, 1, 2, 2}),
+          makeNullableFlatVector<int64_t>({1, std::nullopt, 2, 3, 5}),
+          makeNullableFlatVector<int64_t>({2, std::nullopt, 3, 4, 6}),
+      })};
+  planNode =
+      PlanBuilder(std::make_shared<core::PlanNodeIdGenerator>())
```

Comment:
This plan is common among both tests (minus the input rows). Please abstract it as a variable so that the code is easier to read.